src/Controller/DefaultController.php

```diff
@@ -136,24 +136,32 @@ public function admin()
     /**
      * Get the sitemap.xml containing all dataset urls.
      *
+     * @param FundingOrgFilter $fundingOrgFilter The funding organization filter utility.
+     *
      * @Route("/sitemap.xml", name="pelagos_sitemap")
      *
      * @return StreamedResponse
      */
-    public function showSiteMapXml()
+    public function showSiteMapXml(FundingOrgFilter $fundingOrgFilter)
     {
-        $response = new StreamedResponse(function () {
+        $criteria = array(
+            'availabilityStatus' =>
+            array(
+                DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE,
+                DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE_REMOTELY_HOSTED,
+            )
+        );
 
-            $datasets = $this->getDoctrine()->getRepository(Dataset::class)->findBy(
-                array(
-                    'availabilityStatus' =>
-                    array(
-                        DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE,
-                        DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE_REMOTELY_HOSTED,
-                    )
-                )
-            );
+        if ($fundingOrgFilter->isActive()) {
+            $criteria = array_merge($criteria, array(
+                'researchGroup' =>
+                $fundingOrgFilter->getResearchGroupsIdArray()
+            ));
+        }
+
+        $datasets = $this->getDoctrine()->getRepository(Dataset::class)->findBy($criteria);
```

Comment:
I don't which is the best practice, autowiring DatasetRepository here or using it like this

Reply:
Autowiring is proably better. I'll look into it.